numpy/core/memmap.py

```diff
@@ -209,10 +209,10 @@ def __new__(subtype, filename, dtype=uint8, mode='r+', offset=0,
         import os.path
         try:
             mode = mode_equivalents[mode]
-        except KeyError:
+        except KeyError as e:
             if mode not in valid_filemodes:
                 raise ValueError("mode must be one of %s" %
-                                 (valid_filemodes + list(mode_equivalents.keys())))
+                                 (valid_filemodes + list(mode_equivalents.keys()))) from e
```

Comment:
`KeyError` doesn't provide very useful context here. Better would be to use `from None`, and put `mode` in the message somewhere.

Reply:
Okay, thanks! I am not 100% sure if my update incorporates `mode` the way you intended.